tests/functional/vault/test_permit.py

```diff
@@ -0,0 +1,50 @@
+from brownie import chain
+from eth_account import Account
+from eth_account._utils.structured_data.hashing import hash_domain
+from eth_account.messages import encode_structured_data
+from eth_utils import encode_hex
+
+
+def test_permit(vault):
+    owner = Account.create()
+    spender = Account.create()
+    nonce = vault.nonces(owner.address)
+    expiry = chain[-1].timestamp + 3600
+    amount = 10 ** 21
+    data = {
+        "types": {
+            "EIP712Domain": [
+                {"name": "name", "type": "string"},
+                {"name": "version", "type": "string"},
+                {"name": "chainId", "type": "uint256"},
+                {"name": "verifyingContract", "type": "address"},
+            ],
+            "Permit": [
+                {"name": "owner", "type": "address"},
+                {"name": "spender", "type": "address"},
+                {"name": "amount", "type": "uint256"},
+                {"name": "nonce", "type": "uint256"},
+                {"name": "expiry", "type": "uint256"},
+            ],
+        },
+        "domain": {
+            "name": "Yearn Vault",
+            "version": vault.apiVersion(),
+            "chainId": 1,
```

Comment:
going to fail testing unless you use `web3.eth.chainId()`

Reply:
Broken in ganache:
```
vault.chain_id() = 1
web3.eth.chain_id = 1337
web3.net.version = 1
```

Reply:
Related to this bug https://github.com/trufflesuite/ganache/issues/1643